flink-tests/src/test/java/org/apache/flink/test/classloading/jar/CheckpointedStreamingProgram.java

```diff
@@ -57,7 +59,7 @@ public static void main(String[] args) throws Exception {
 
 
 	// with Checkpoining
-	public static class SimpleStringGenerator implements SourceFunction<String>, Checkpointed<Integer> {
```

Comment:
I understand why you did not replace ´Checkpointed` with a different interface, but this technically changes the test. If a function can still implement these interfaces without doing anything within snapshot/restore, then we should cover this with tests.